dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -324,222 +395,17 @@ public void setInterceptorChain(InterceptorChain ic) {
-                if (serviceRedirection != NOT_FOUND) {
-                    return serviceRedirection;
-                }
-            } else if (key == HTTP_SERVLET_RESPONSE) {
-                if (httpServletResponse != NOT_FOUND) {
-                    return httpServletResponse;
-                }
-            } else if (key == RESOURCE_METHOD) {
-                if (resourceMethod != NOT_FOUND) {
-                    return resourceMethod;
-                }
-            } else if (key == ONE_WAY_REQUEST) {
-                if (oneWayRequest != NOT_FOUND) {
-                    return oneWayRequest;
-                }
-            } else if (key == ASYNC_RESPONSE) {
-                if (asyncResponse != NOT_FOUND) {
-                    return asyncResponse;
-                }
-            } else if (key == THREAD_CONTEXT_SWITCHED) {
-                if (threadContextSwitched != NOT_FOUND) {
-                    return threadContextSwitched;
-                }
-            } else if (key == OutgoingChainInterceptor.CACHE_INPUT_PROPERTY) {
-                if (cacheInputProperty != NOT_FOUND) {
-                    return cacheInputProperty;
-                }
-            } else if (key == PhaseInterceptorChain.PREVIOUS_MESSAGE) {
-                if (previousMessage != NOT_FOUND) {
-                    return previousMessage;
-                }
-            } else if (key == AbstractHTTPDestination.RESPONSE_HEADERS_COPIED) {
-                if (responseHeadersCopied != NOT_FOUND) {
-                    return responseHeadersCopied;
-                }
-            } else if (key == SSE_EVENT_SINK) {
-                if (sseEventSink != NOT_FOUND) {
-                    return sseEventSink;
-                }
-            } else if (key == REQUESTOR_ROLE) {
-                if (requestorRole != NOT_FOUND) {
-                    return requestorRole;
-                }
-            } else if (key == PARTIAL_RESPONSE_MESSAGE) {
-                if (partialResponse != NOT_FOUND) {
-                    return partialResponse;
-                }
-            } else if (key == EMPTY_PARTIAL_RESPONSE_MESSAGE) {
-                if (emptyPartialResponse != NOT_FOUND) {
-                    return emptyPartialResponse;
-                }
-            } else if (key == ENDPOINT_ADDRESS) {
-                if (endpointAddress != NOT_FOUND) {
-                    return endpointAddress;
-                }
-            } else if (key == INBOUND_MESSAGE) {
-                if (inboundMessage != NOT_FOUND) {
-                    return inboundMessage;
-                }
-            }
+        Integer index = KEYMAP.getOrDefault(key, KEY_NOT_FOUND);
+        if (index != KEY_NOT_FOUND) {
+            Object value = propertyValues[index];
+            return value == NOT_FOUND ? null : value;
         }
         //Liberty code change end
 
-        o = getOrDefault(key, NOT_FOUND);
+        Object o = getOrDefault(key, NOT_FOUND);
```

Comment:
Either this should call super.getOrDefault or this whole method should just call getOrDefault to use the impl that is in this class.